tests/test_server.py

```diff
@@ -474,3 +475,59 @@ def test_streaming_azure(client):
             assert (
                 actual_output == expected_output
             ), "adala did not return expected output"
+
+
+# can't use vcr here because server makes async requests
+@pytest.mark.use_openai
```

Comment:
it is not clear why responses can't be mocked here. I'd prefer to avoid manual pytest runs and use it alongside other checks, if it is currently implied by use_openai mark here.

Reply:
Don't need this test as it almost duplicates the `test_refine_skill`, will put more variations there and remove this one